src/_pytest/mark/structures.py

```diff
@@ -14,6 +14,11 @@
 EMPTY_PARAMETERSET_OPTION = "empty_parameter_set_mark"
 
 
+class ParametrizeError(Exception):
```

Comment:
When this is raised this will produce a very long traceback with the pytest internals:

```
============================== ERRORS ===============================
____________ ERROR collecting test-empty-parametrize.py _____________
.env36\lib\site-packages\pluggy\hooks.py:258: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.env36\lib\site-packages\pluggy\manager.py:67: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.env36\lib\site-packages\pluggy\manager.py:61: in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
src\_pytest\python.py:255: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
src\_pytest\python.py:440: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
.env36\lib\site-packages\pluggy\hooks.py:258: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.env36\lib\site-packages\pluggy\manager.py:67: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.env36\lib\site-packages\pluggy\manager.py:61: in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
src\_pytest\python.py:164: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
src\_pytest\python.py:975: in parametrize
    function_definition=self.definition,
src\_pytest\mark\structures.py:140: in _for_parametrize
    mark = get_empty_parameterset_mark(config, argnames, func)
src\_pytest\mark\structures.py:47: in get_empty_parameterset_mark
    raise ParametrizeError("Empty parameter set in '%s'" % f_name)
E   _pytest.mark.structures.ParametrizeError: Empty parameter set in 'test'
!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!
====================== 1 error in 0.17 seconds ======================
```

but if we raise `Collector.CollectError` we will much nicer representation:

```
============================== ERRORS ===============================
____________ ERROR collecting test-empty-parametrize.py _____________
Empty parameter set in 'test'
!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!
====================== 1 error in 0.08 seconds ======================
```

Reply:
good find, lets use that

Reply:
Excellent suggestion, of course I had no clue of that, thanks! It's coming...